eclass/kernel-build.eclass

```diff
@@ -350,7 +350,15 @@ kernel-build_src_install() {
 	dosym "../../../${kernel_dir}" "/lib/modules/${module_ver}/source"
 
 	if [[ ${KERNEL_IUSE_SECUREBOOT} ]]; then
-		secureboot_sign_efi_file "${ED}${kernel_dir}/${image_path}"
+		# Arm64 has no compressed kernel support, instead the DOS header we
+		# add the signature to is in the image itself, so uncompress first.
+		if use arm64 && [[ ${image_path} == *".gz" ]]; then
```

Comment:
arm64 has well-defined image path, so these two are redundant. Let's wait for the answer from riscv folks, and perhaps keep just `*.gz` part if riscv is also affected. Also no quoting needed here, and it makes it look weird.

Reply:
I still can't find any secure boot signing documentation for riscv. But judging from the `Makefile` we are going to have the same problem on riscv:

On x86 we have:
```
KBUILD_IMAGE := $(boot)/bzImage
```

On arm we have:
```
ifeq ($(CONFIG_XIP_KERNEL),y)
KBUILD_IMAGE := $(boot)/xipImage
else
KBUILD_IMAGE := $(boot)/zImage
endif
```

On arm64 we have:
```
ifeq ($(CONFIG_EFI_ZBOOT),)
KBUILD_IMAGE	:= $(boot)/Image.gz
else
KBUILD_IMAGE	:= $(boot)/vmlinuz.efi
endif
```

And on riscv we have:
```
ifeq ($(CONFIG_XIP_KERNEL),y)
KBUILD_IMAGE := $(boot)/xipImage
else
KBUILD_IMAGE	:= $(boot)/Image.gz
endif
```

I'm gonna look into what exactly this `XIP_KERNEL` is and whether we maybe want to enable `CONFIG_EFI_ZBOOT` on arm64 instead when we have `USE=secureboot`.

Reply:
@mgorny what do you think about giving `CONFIG_EFI_ZBOOT` another try? It was [disabled](https://bugs.gentoo.org/897684) in February because it didn't work with qemu, but it seems that this might now work with [qemu 8](https://wiki.qemu.org/ChangeLog/8.0) and up which was released in April.

For ZBOOT we have to sign both the decompressor and the kernel itself, luckily the kernel build system can do this for us. I put this behind `use secureboot` so it doesn't effect the older versions and this also provides a convenient way to make the name of the image conditional in `dist-kernel-utils.eclass`.

We are going to have to keep the `if [[ ${image_path} == *.gz ]]` logic around for riscv, at least until riscv gains something similar to zboot. I don't really like this because it assumes the kernel is gzip compressed which is something the user could override in `/etc/kernel/config.d` this would then break `USE=secureboot`. Alternatively (provided the zboot thing works) we can remove the `if [[ ${image_path} == *.gz ]]` and `package.use.mask` the secureboot flag on riscv.

I tested this on my amd64 machine, but of course none of this is relevant on amd64 so it still needs a proper test on arm64 (@gentoo/arm64 ) and it would be nice to also test this on riscv (@gentoo/risc-v ). Easiest way for a quick test would be to take any kernel with signed modules support and set `SECUREBOOT_SIGN_KEY` and `SECUREBOOT_SIGN_CERT` to the same key (e.g.`/usr/src/linux/certs/signing_key.pem`).

Reply:
> We are going to have to keep the `if [[ ${image_path} == *.gz ]]` logic around for riscv, at least until riscv gains something similar to zboot. I don't really like this because it assumes the kernel is gzip compressed which is something the user could override in `/etc/kernel/config.d` this would then break `USE=secureboot`. Alternatively (provided the zboot thing works) we can remove the `if [[ ${image_path} == *.gz ]]` and `package.use.mask` the secureboot flag on riscv.

Actually, I was wrong, riscv does support zboot. So we can apply the same fix as for arm64 and get rid of the sketchy gunzip/gzip. The question that remains is if zboot actually works in the `src_test`.